dependencies.yaml

```diff
@@ -100,22 +210,50 @@ dependencies:
     common:
       - output_types: [conda, requirements]
         packages:
-          - librmm=23.04.*
+          - librmm==23.4.*
       - output_types: conda
         packages:
           - librdkafka=1.7.0
-  build_python:
+          # Hard pin the patch version used during the build, must match the CMake version.
+          - libarrow==10.0.1.*
```

Comment:
This is the version that is used in [our CMake code that searches for arrow libs to build against](https://github.com/rapidsai/cudf/blob/branch-23.04/cpp/cmake/thirdparty/get_arrow.cmake#L382). Due to the way that arrow's CMake is set up, we need an exact match to the patch version or CMake will clone a copy of the arrow repo instead of using a preexisting lib. As a result, the build-time pin needs to be exact down to the patch version.

Reply:
Perhaps add cross-links in the comments (here) pointing at that file, and in `get_arrow.cmake` pointing here?

Reply:
Good call, adding comments to indicate that these two need to be updated in sync is probably helpful (and will be even more important for other libraries beyond libcudf that have even more libraries used for both C++ and Python builds).